emu/src/cpu/arm7tdmi.rs

```diff
@@ -686,29 +686,63 @@ impl Arm7tdmi {
         let rs_hs = op_code.get_bits(3..=5);
         let rd_hd = op_code.get_bits(0..=2);
 
-        let _destination_register: usize = (h1 as u16 * 8 + rd_hd).try_into().unwrap();
+        if !h1 && !h2 && (op == 0b00 || op == 0b01 || op == 0b10) {
+            panic!("H1=0 H2=0 is not supported with ADD, CMP, MOV");
+        } else if h1 && op == 0b11 {
+            panic!("H1=1 is not supported with BX")
+        }
+
+        let destination_register: usize = (h1 as u16 * 8 + rd_hd).try_into().unwrap();
         let source_register: usize = (h2 as u16 * 8 + rs_hs).try_into().unwrap();
 
+        let first_op = self.registers.register_at(destination_register)
+            + match destination_register as u32 {
+                REG_PROGRAM_COUNTER => 4,
+                _ => 0,
+            };
+
+        let second_op = self.registers.register_at(source_register)
+            + match source_register as u32 {
+                REG_PROGRAM_COUNTER => 4,
+                _ => 0,
+            };
+
         match op {
             // Add
-            0b00 => todo!(), //Some(SIZE_OF_THUMB_INSTRUCTION)
+            0b00 => {
+                self.registers
+                    .set_register_at(destination_register, first_op + second_op);
```

Comment:
Simple `+` or `add_inner`?

Reply:
Simple `+` is ok since only MOV changes CPSR in this instruction